src/main/java/de/rabea/game/Board.java

```diff
@@ -29,10 +29,21 @@ public void placeMark(int position, Mark mark) {
         cells[position] = mark;
     }
 
+    public Board placeMarkOnNewBoard(int position, Mark mark, Board board) {
+
+        Board newBoard = board.copy();
```

Comment:
Do you still need the copy method?
You could make a copy of the marks array, modify it where needed and call the ```new Board(Mark [] marks)``` constructor.